### DISCUSSION TITLE
Block bounding boxes

### DISCUSSION THREAD

**Discussion by jorisguffens** [Minestom/Minestom#618] [Help] [answered]
Is there a way to get the bounding box of blocks? This would be very useful for non-cubic blocks like slabs, stairs, cauldron...

Also, are there any plans to update the physics/collisions to respect these blocks? Entities now appear floating when walking on lower height blocks.

![image](https://user-images.githubusercontent.com/8937042/151676760-0aca447b-22d4-4045-aa3b-39b9d7c5691a.png)

**Comment by TheMode** [comment, marked as answer]
There is a WIP branch adding support for special bounding boxes https://github.com/Minestom/Minestom/pull/523

Block shapes are not exposed directly in the API (as the format may change), but can be retrieved from the [block registry](https://github.com/Minestom/MinestomDataGenerator/blob/master/DataGenerator/src/main/java/net/minestom/generators/BlockGenerator.java) and accessible using `Block#registry#getString`. Notice the json files in your final jar archive.
